components/ai_chat/core/browser/engine/engine_consumer_conversation_api.cc

```diff
@@ -143,13 +144,17 @@ void EngineConsumerConversationAPI::GenerateRewriteSuggestion(
 }
 
 void EngineConsumerConversationAPI::GenerateQuestionSuggestions(
-    const bool& is_video,
-    const std::string& page_content,
+    PageContentses page_contents,
     const std::string& selected_language,
     SuggestedQuestionsCallback callback) {
   std::vector<ConversationEvent> conversation;
-  conversation.emplace_back(
-      GetAssociatedContentConversationEvent(page_content, is_video));
+  int remaining_length = max_associated_content_length_;
+  for (const auto& content : page_contents) {
+    conversation.emplace_back(
+        GetAssociatedContentConversationEvent(content, remaining_length));
+    remaining_length -= content.get().content.size();
```

Comment:
This can end up calling GetAssociatedContentConversationEvent with negative length.
Can we accumulate current content length and include the content until it exceed `max_associated_content_length_`?

Reply:
Added a `break` if the `remaining_length` hits zero

Reply:
we can easily check if is sufficient to be subtracted or just 0 and don't use int.